### PR TITLE
CBMC: Correct type used in mqtt_state for operation list allocation.

### PR DESCRIPTION
**Pull request by markrtuttle**
This patch repairs a type used in operation list allocation for the CBMC proofs.  Since the offset of the link struct in the two types are both 0, this won't change any results.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cbmc/proofs/mqtt_state.c [hunk 1]
```diff
@@ -182,7 +182,7 @@ bool valid_IotMqttOperationList( const IotListDouble_t *pOp,
   IotListDouble_t *pLink;
   IotContainers_ForEach( pOp, pLink ) {
     IotMqttOperation_t
-      *pElt = IotLink_Container( _mqttSubscription_t, pLink, link );
+      *pElt = IotLink_Container( struct _mqttOperation, pLink, link );
```
Comment (feliperodri) [inline cbmc/proofs/mqtt_state.c]: You also need to update line 460.

Comment (markrtuttle) [reply cbmc/proofs/mqtt_state.c]: Why?  Line 460 is iterating over a subscription list and the type is correct.  This mistake was made iterating over an operation list.

Comment (feliperodri) [reply cbmc/proofs/mqtt_state.c]: Oh true! Approved!

Comment (gordonwang0) [reply cbmc/proofs/mqtt_state.c]: Spacing of these 3 lines seems inconsistent.

Comment (markrtuttle) [reply cbmc/proofs/mqtt_state.c]: I'm not sure how to break the long line, I pushed a revision that doesn't break the line.

**Pull request merged**